waku/node/waku_node.nim

```diff
@@ -431,9 +464,76 @@ proc filterSubscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], contentT
     for pubsub, topics in topicMap.pairs:
       info "subscribed to topic", pubsubTopic=pubsub, contentTopics=topics
 
-proc filterUnsubscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], contentTopics: ContentTopic|seq[ContentTopic],
-                  peer: RemotePeerInfo|string) {.async, gcsafe, raises: [Defect, ValueError].} =
-  ## Unsubscribe from a content filter.
+proc filterV2Subscribe*(node: WakuNode,
+                        pubsubTopic: Option[PubsubTopic],
+                        contentTopics: ContentTopic|seq[ContentTopic],
+                        peer: RemotePeerInfo|string):
+
+                Future[FilterSubscribeResult]
+
+                {.async, gcsafe, raises: [Defect, ValueError].} =
+
+  ## Registers for messages that match a specific filter. Triggers the handler whenever a message is received.
+  if node.wakuFilterClient.isNil():
+    error "cannot register filter subscription to topic", error="waku filter client is not set up"
+    return err(FilterSubscribeError.serviceUnavailable())
+
+  let remotePeerRes = parsePeerInfo(peer)
+  if remotePeerRes.isErr():
+    error "Couldn't parse the peer info properly", error = remotePeerRes.error
```

Comment:
Something that we should enhance in the future is to make the `waku_node` trace the minimum possible (or nothing ideally)

From my point of view, we should always deal with `Result[T]` types and leave the traces for the upper layers. There shouldn't be any trace/log in `waku_node`. wdyt about this statement @jm-clius ?